isce2_topsapp/delivery_prep.py

```diff
@@ -113,37 +114,36 @@ def gen_browse_imagery(nc_path: Path,
     return out_path
 
 
-def format_time_string(time_str: str) -> str:
-    """Generates formatted string for ingest schema using built in python manipulations.
-    New ASF API provides time string as 2022-05-20T07:11:01.000Z and we need 6 decimal places
-    at end of string, specifically, 2022-05-20T07:11:01.000000Z
-    """
-    assert time_str[-1] == 'Z'
-
-    # removes Z at end and focuses in on decimal
-    temp = time_str[:-1].split('.')
-
-    # update the split string with required decimals
-    temp[-1] = f'{int(temp[-1]):06d}'
-
-    # rejoin and return
-    formatted_time_str = '.'.join(temp)
-    return f'{formatted_time_str}Z'
-
-
 def format_metadata(nc_path: Path,
                     all_metadata: dict) -> dict:
 
     now = datetime.datetime.now()
     label = nc_path.name[:-3]  # removes suffix .nc
     geojson = all_metadata['gunw_geo'].__geo_interface__
 
-    ref_props = all_metadata['reference_properties'][0]
-    sec_props = all_metadata['secondary_properties'][0]
+    ref_props_all = sorted(all_metadata['reference_properties'],
+                           key=lambda prop: prop['startTime'])
+    ref_props_first = ref_props_all[0]
+    sec_props_all = sorted(all_metadata['secondary_properties'],
+                           key=lambda prop: prop['startTime'])
+    sec_props_first = sec_props_all[0]
     b_perp = read_baseline_perp(nc_path).mean()
 
-    startTime_f = format_time_string(ref_props["startTime"])
-    stopTime_f = format_time_string(ref_props["stopTime"])
+    ref_start_times = [parse(props['startTime']) for props in ref_props_all]
+    ref_stop_times = [parse(props['stopTime']) for props in ref_props_all]
+    sec_start_times = [parse(props['startTime']) for props in sec_props_all]
+
+    ref_start_time = ref_start_times[0]
+    ref_stop_time = ref_stop_times[-1]
+    sec_start_time = sec_start_times[0]
+
+    # The %f is miliseconds zero-padded with 6 decimals - just as we need!
+    ref_start_time_formatted = ref_start_time.strftime('%Y-%m-%dT%H:%M:%S.%fZ')
+    ref_stop_time_formatted = ref_stop_time.strftime('%Y-%m-%dT%H:%M:%S.%fZ')
+
+    # We want the nearest day (dt.days takes a floor) so we use total seconds and then round
+    temporal_baseline_seconds = (ref_start_time - sec_start_time).total_seconds()
+    temporal_baseline_days = round(temporal_baseline_seconds / 60 / 60 / 24)
```

Comment:
Providing some context for @dbekaert important comments about obtaining an accurate temporal baseline in days. We expect this to be a multiple of 6 based on S1's repeat pass frequency.

When the temporal baseline (in days) is for example 5.999, the if `delta` is a `datetime.timedelta` object, then `delta.days` will be 5 (not 6). Therefore, we want to do the rounding using `total_seconds()`. This resolves David's concerns.

To be clear, this concern can occur anywhere, not just at a pass that passes through midnight in UTC.